src/events/interactionCreate.ts

```diff
@@ -0,0 +1,18 @@
+import { ButtonInteraction, CommandInteraction, Interaction } from "discord.js";
+import { ctx } from "../ctx";
+
+export default async function interactionCreate(interaction: Interaction) {
+	if (interaction.isCommand()) {
+		await handleCommandInteractions(interaction);
+	} else if (interaction.isButton()) {
+		await handleButtonInteractions(interaction);
+	}
```

Comment:
Maybe wrap in some error handling, also are these functions really necessary? Both are a single line of code.

Reply:
we have a client.on("error") funciton

Reply:
And even if it did, the user wouldn't be aware of it

Reply:
i actually asked my superior about it (bc clean code and stuff), and their opinion is to keep those as seperate functions